spyder_kernels/utils/nsview.py

```diff
@@ -445,6 +449,8 @@ def value_to_display(value, minmax=False, level=0):
         elif (isinstance(value, datetime.date) or
               isinstance(value, datetime.timedelta)):
             display = str(value)
+        elif isinstance(value, NUMERIC_TYPES) and numeric_value:
+            display = value
         elif (isinstance(value, NUMERIC_TYPES) or
               isinstance(value, bool) or
               isinstance(value, numeric_numpy_types)):
```

Comment:
I think adding `numeric_value` is not necessary if you simply rewrite these lines like this:

```python
elif (isinstance(value, NUMERIC_TYPES) or 
      isinstance(value, numeric_numpy_types)):
    display = value
elif isinstance(value, bool):
    display = repr(value)
```

I included `numeric_numpy_types` so people can sort them correctly too.

Reply:
There's an important thing I just verified about this: `complex` is in `NUMERIC_TYPES`, but complex numbers can't be sorted against other numbers (be them complex or not):

```
(2 + 3j) > 0
TypeError: '>' not supported between instances of 'complex' and 'int'

(2 + 3j) > (3 + 2j) 
TypeError: '>' not supported between instances of 'complex' and 'complex'
```

However, Numpy complex types can be sorted correctly. I also verified that there are no problems with `np.bool_`. So, I think we need to remove `complex` numbers at this point.

Reply:
> 
> 
> I think adding `numeric_value` is not necessary if you simply rewrite these lines like this:
> 
> ```python
> elif (isinstance(value, NUMERIC_TYPES) or 
>       isinstance(value, numeric_numpy_types)):
>     display = value
> elif isinstance(value, bool):
>     display = repr(value)
> ```
> 
> I included `numeric_numpy_types` so people can sort them correctly too.

The reason to add the `numeric_value` param is that without it the value shown for the list in the Variable Explorer changes to `list object` instead of the current representation that is `[1e+16, 10.0, 1.0, 0.1, 1e-06, 0.0, -1e-06, -1.0, -10.0, -1e+16]` (`collections_display` uses `value_to_display` to get the representation of each element of the collection/list so that was causing the representation to change when I tested a simpler approach as the suggested)

Reply:
> 
> 
> There's an important thing I just verified about this: `complex` is in `NUMERIC_TYPES`, but complex numbers can't be sorted against other numbers (be them complex or not):
> 
> ```
> (2 + 3j) > 0
> TypeError: '>' not supported between instances of 'complex' and 'int'
> 
> (2 + 3j) > (3 + 2j) 
> TypeError: '>' not supported between instances of 'complex' and 'complex'
> ```
> 
> However, Numpy complex types can be sorted correctly. I also verified that there are no problems with `np.bool_`. So, I think we need to remove `complex` numbers at this point.

Should I remove `complex` from the `NUMERIC_TYPES` constant then?

Reply:
> The reason to add the numeric_value param is that without it the value shown for the list in the Variable Explorer changes to list object instead of the current representation that is [1e+16, 10.0, 1.0, 0.1, 1e-06, 0.0, -1e-06, -1.0, -10.0, -1e+16] (collections_display uses value_to_display to get the representation of each element of the collection/list so that was causing the representation to change when I tested a simpler approach as the suggested)

Ok, understood.

> Should I remove complex from the NUMERIC_TYPES constant then?

Yeah, I think that's fine.